--- a/tools/build/v2/tools/darwin.jam	2011-06-06 22:36:21.000000000 +0200
+++ b/tools/build/v2/tools/darwin.jam	2012-03-15 18:14:22.000000000 +0100
@@ -70,7 +70,7 @@
 #
 #   <root>PATH
 #       Platform root path. The common autodetection will set this to
-#       "/Developer". And when a command is given it will be set to
+#       "/Applications/Xcode.app/Contents/Developer". And when a command is given it will be set to
 #       the corresponding "*.platform/Developer" directory.
 #
 rule init ( version ? : command * : options * : requirement * )
@@ -104,7 +104,7 @@
         bin ?= [ common.get-absolute-tool-path $(command[1]) ] ;
         if $(bin) = "/usr/bin"
         {
-            root ?= /Developer ;
+            root ?= /Applications/Xcode.app/Contents/Developer ;
         }
         else
         {
@@ -312,9 +312,13 @@
 # Determine the MacOSX SDK versions installed and their locations.
 local rule init-available-sdk-versions ( condition * : root ? )
 {
-    root ?= /Developer ;
-    local sdks-root = $(root)/SDKs ;
-    local sdks = [ GLOB $(sdks-root) : MacOSX*.sdk iPhoneOS*.sdk iPhoneSimulator*.sdk ] ;
+    root ?= /Applications/Xcode.app/Contents/Developer ;
+    local platforms = [ GLOB /Applications/Xcode.app/Contents/Developer/Platforms : *.platform ] ;
+    local sdks = ;
+    for local platform in $(platforms)
+    {
+        sdks += [ GLOB $(platform)/Developer/SDKs : *.sdk ] ;
+    }
     local result ;
     for local sdk in $(sdks)
     {
