--- CMakeLists.txt.old	2012-02-01 21:45:51.734378620 -0700
+++ CMakeLists.txt	2012-02-01 21:46:08.113080799 -0700
@@ -20,6 +20,14 @@
   endif()
 endif()
 
+
+# make sure -fPIC is used, this is a patch provided by openstudio
+if (UNIX)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -fno-strict-aliasing")
+  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC -fno-strict-aliasing")
+endif()
+
+
 # Set up standard install locations
 set(INC_INSTALL_DIR "include/vtkCharts")
 set(LIB_INSTALL_DIR "lib")
