diff --git a/enc/make_encmake.rb b/enc/make_encmake.rb
index 6ea2213..9b20dee 100755
--- a/enc/make_encmake.rb
+++ b/enc/make_encmake.rb
@@ -43,7 +43,7 @@ end
 
 ALPHANUMERIC_ORDER = proc {|e| e.scan(/(\d+)|(\D+)/).map {|n,a| a||[n.size,n.to_i]}.flatten}
 def target_encodings
-  encs = Dir.open($srcdir) {|d| d.grep(/.+\.c\z/)} - BUILTIN_ENCS - ["mktable.c"]
+  encs = Dir.open($srcdir) {|d| d.grep(/.+\.c\z/)} - BUILTIN_ENCS - ["mktable.c", "encinit.c"]
   encs.each {|e| e.chomp!(".c")}
   encs.reject! {|e| !ENC_PATTERNS.any? {|p| File.fnmatch?(p, e)}} if !ENC_PATTERNS.empty?
   encs.reject! {|e| NOENC_PATTERNS.any? {|p| File.fnmatch?(p, e)}}
diff --git a/ext/Setup b/ext/Setup
index 98ffffd..688219c 100644
--- a/ext/Setup
+++ b/ext/Setup
@@ -1,28 +1,28 @@
-#option nodynamic
+option nodynamic
 
-#Win32API
-#bigdecimal
-#dbm
-#digest
-#digest/md5
-#digest/rmd160
-#digest/sha1
-#digest/sha2
-#etc
-#fcntl
-#gdbm
-#io/wait
-#nkf
-#openssl
-#pty
-#racc/cparse
-#readline
-#ripper
-#sdbm
-#socket
-#stringio
-#strscan
-#syslog
-#tk
+Win32API
+bigdecimal
+dbm
+digest
+digest/md5
+digest/rmd160
+digest/sha1
+digest/sha2
+etc
+fcntl
+gdbm
+io/wait
+nkf
+openssl
+pty
+racc/cparse
+readline
+ripper
+sdbm
+socket
+stringio
+strscan
+syslog
+tk
 #win32ole
-#zlib
+zlib
