--- a/win32/win32.c	2019-01-09 20:30:35.037112700 -0800
+++ b/win32/win32.c	2019-01-09 20:31:11.825873700 -0800
@@ -6212,6 +6212,7 @@
 
     share_delete = oflag & O_SHARE_DELETE ? FILE_SHARE_DELETE : 0;
     oflag &= ~O_SHARE_DELETE;
+    /*
     if ((oflag & O_TEXT) || !(oflag & O_BINARY)) {
 	fd = _wopen(file, oflag, pmode);
 	if (fd == -1) {
@@ -6226,7 +6227,7 @@
 	}
 	return fd;
     }
-
+    */
     sec.nLength = sizeof(sec);
     sec.lpSecurityDescriptor = NULL;
     if (oflag & O_NOINHERIT) {
