diff --git a/ext/Setup b/ext/Setup
index ac79c86..39fb0f0 100644
--- a/ext/Setup
+++ b/ext/Setup
@@ -1,43 +1,43 @@
-#option nodynamic
+option nodynamic
 
-#bigdecimal
-#cgi/escape
-#continuation
-#coverage
-#date
-#dbm
-#digest/bubblebabble
-#digest
-#digest/md5
-#digest/rmd160
-#digest/sha1
-#digest/sha2
-#etc
-#fcntl
-#fiber
-#fiddle
-#gdbm
-#io/console
-#io/nonblock
-#io/wait
-#json
-#json/generator
-#json/parser
-#nkf
-#objspace
-#openssl
-#pathname
-#psych
-#pty
-#racc/cparse
-#rbconfig/sizeof
-#readline
-#ripper
-#sdbm
-#socket
-#stringio
-#strscan
-#syslog
-#win32
-#win32ole
-#zlib
+bigdecimal
+cgi/escape
+continuation
+coverage
+date
+dbm
+digest/bubblebabble
+digest
+digest/md5
+digest/rmd160
+digest/sha1
+digest/sha2
+etc
+fcntl
+fiber
+fiddle
+gdbm
+io/console
+io/nonblock
+io/wait
+json
+json/generator
+json/parser
+nkf
+objspace
+openssl
+pathname
+psych
+pty
+racc/cparse
+rbconfig/sizeof
+readline
+ripper
+sdbm
+socket
+stringio
+strscan
+syslog
+win32
+win32ole
+zlib
diff --git a/ext/zlib/extconf.rb b/ext/zlib/extconf.rb
index fd2f168..ba63d9b 100644
--- a/ext/zlib/extconf.rb
+++ b/ext/zlib/extconf.rb
@@ -11,7 +11,7 @@
 dir_config 'zlib'
 
 
-if %w'z libz zlib1 zlib zdll zlibwapi'.find {|z| have_library(z, 'deflateReset')} and
+if %w'z zlibstatic zlibstaticd'.find {|z| have_library(z, 'deflateReset', 'zlib.h')} and
     have_header('zlib.h') then
   have_zlib = true
 else
diff --git a/lib/mkmf.rb b/lib/mkmf.rb
index 7f53bb4..20ac890 100644
--- a/lib/mkmf.rb
+++ b/lib/mkmf.rb
@@ -2708,7 +2708,21 @@ def MAIN_DOES_NOTHING(*refs)
   ##
   # A C main function which does no work
 
-  MAIN_DOES_NOTHING = config_string('MAIN_DOES_NOTHING') || "int main(int argc, char **argv)\n{\n  return 0;\n}"
+  MAIN_DOES_NOTHING = config_string('MAIN_DOES_NOTHING') || <<-SRC
+int main(int argc, char **argv)
+{
+  return 0;
+}
+
+int rb_hasFile(const char *t_filename) {
+  return 0;
+}
+
+int rb_require_embedded(const char *t_filename) {
+  return 0;
+}
+SRC
+
   UNIVERSAL_INTS = config_string('UNIVERSAL_INTS') {|s| Shellwords.shellwords(s)} ||
     %w[int short long long\ long]
 
diff --git a/lib/rubygems.rb b/lib/rubygems.rb
index 2762bfc..2280a9e 100644
--- a/lib/rubygems.rb
+++ b/lib/rubygems.rb
@@ -1354,24 +1354,24 @@ def clear_default_specs
 
   MARSHAL_SPEC_DIR = "quick/Marshal.#{Gem.marshal_version}/"
 
-  autoload :BundlerVersionFinder, 'rubygems/bundler_version_finder'
-  autoload :ConfigFile,         'rubygems/config_file'
-  autoload :Dependency,         'rubygems/dependency'
-  autoload :DependencyList,     'rubygems/dependency_list'
-  autoload :DependencyResolver, 'rubygems/resolver'
-  autoload :Installer,          'rubygems/installer'
-  autoload :Licenses,           'rubygems/util/licenses'
-  autoload :PathSupport,        'rubygems/path_support'
-  autoload :Platform,           'rubygems/platform'
-  autoload :RequestSet,         'rubygems/request_set'
-  autoload :Requirement,        'rubygems/requirement'
-  autoload :Resolver,           'rubygems/resolver'
-  autoload :Source,             'rubygems/source'
-  autoload :SourceList,         'rubygems/source_list'
-  autoload :SpecFetcher,        'rubygems/spec_fetcher'
-  autoload :Specification,      'rubygems/specification'
-  autoload :Util,               'rubygems/util'
-  autoload :Version,            'rubygems/version'
+  require 'rubygems/bundler_version_finder'
+  require 'rubygems/config_file'
+  require 'rubygems/dependency'
+  require 'rubygems/dependency_list'
+  require 'rubygems/resolver'
+  require 'rubygems/installer'
+  require 'rubygems/util/licenses'
+  require 'rubygems/path_support'
+  require 'rubygems/platform'
+  require 'rubygems/request_set'
+  require 'rubygems/requirement'
+  require 'rubygems/resolver'
+  require 'rubygems/source'
+  require 'rubygems/source_list'
+  require 'rubygems/spec_fetcher'
+  require 'rubygems/specification'
+  require 'rubygems/util'
+  require 'rubygems/version'
 
   require "rubygems/specification"
 end
diff --git a/lib/rubygems/version.rb b/lib/rubygems/version.rb
index ff5c1c1..ff161af 100644
--- a/lib/rubygems/version.rb
+++ b/lib/rubygems/version.rb
@@ -150,7 +150,7 @@
 # a zero to give a sensible result.
 
 class Gem::Version
-  autoload :Requirement, 'rubygems/requirement'
+  require 'rubygems/requirement'
 
   include Comparable
 
diff --git a/load.c b/load.c
index cfc8b1f..19a5342 100644
--- a/load.c
+++ b/load.c
@@ -8,6 +8,9 @@
 #include "eval_intern.h"
 #include "probes.h"
 
+int rb_hasFile(const char *t_filename);
+int rb_require_embedded(const char *t_filename);
+
 static VALUE ruby_dln_librefs;
 
 #define IS_RBEXT(e) (strcmp((e), ".rb") == 0)
@@ -1053,8 +1056,12 @@ rb_require_safe(VALUE fname, int safe)
 VALUE
 rb_require(const char *fname)
 {
+  if ( rb_hasFile(fname) ) {
+    return rb_require_embedded(fname);
+  } else {
     VALUE fn = rb_str_new_cstr(fname);
     return rb_require_safe(fn, rb_safe_level());
+  }
 }
 
 static int
diff --git a/main.c b/main.c
index 6e8ab7b..7e1b2a3 100644
--- a/main.c
+++ b/main.c
@@ -42,3 +42,12 @@ main(int argc, char **argv)
 	return ruby_run_node(ruby_options(argc, argv));
     }
 }
+
+int rb_hasFile(const char *t_filename) {
+  return 0;
+}
+
+int rb_require_embedded(const char *t_filename) {
+  return 0;
+}
+
