diff -r -u VTK/CMakeLists.txt VTK-new/CMakeLists.txt
--- VTK/CMakeLists.txt	2011-04-26 07:55:09.000000000 -0600
+++ VTK-new/CMakeLists.txt	2012-05-12 09:47:22.000000000 -0600
@@ -28,6 +28,14 @@
   unset(WIN32)
 ENDIF()
 
+
+# make sure -fPIC is used, this is a patch provided by openstudio
+if (UNIX)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -fno-strict-aliasing")
+  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC -fno-strict-aliasing")
+endif()
+
+
 # the following lines are for cross compiling support
 # we may get here also from ParaView, in this case don't change the filename
 IF(NOT EXPORT_EXECUTABLES_FILE)
Only in VTK-new/Examples: ._.DS_Store
diff -r -u VTK/GUISupport/Qt/vtkQtSQLDatabase.cxx VTK-new/GUISupport/Qt/vtkQtSQLDatabase.cxx
--- VTK/GUISupport/Qt/vtkQtSQLDatabase.cxx	2011-04-26 07:55:17.000000000 -0600
+++ VTK-new/GUISupport/Qt/vtkQtSQLDatabase.cxx	2012-05-12 09:47:22.000000000 -0600
@@ -19,7 +19,7 @@
 -------------------------------------------------------------------------*/
 
 // Check for Qt SQL module before defining this class.
-#include <qglobal.h>
+#include <QtCore/qglobal.h>
 #if (QT_EDITION & QT_MODULE_SQL)
 
 #include "vtkQtSQLDatabase.h"
diff -r -u VTK/GUISupport/Qt/vtkQtSQLQuery.cxx VTK-new/GUISupport/Qt/vtkQtSQLQuery.cxx
--- VTK/GUISupport/Qt/vtkQtSQLQuery.cxx	2011-04-26 07:55:17.000000000 -0600
+++ VTK-new/GUISupport/Qt/vtkQtSQLQuery.cxx	2012-05-12 10:11:51.000000000 -0600
@@ -19,7 +19,7 @@
 -------------------------------------------------------------------------*/
 
 // Check for Qt SQL module before defining this class.
-#include <qglobal.h>
+#include <QtCore/qglobal.h>
 #if (QT_EDITION & QT_MODULE_SQL)
 
 #include "vtkQtSQLQuery.h"
