--- CMakeLists.txt.old	2012-02-01 21:09:00.114213105 -0700
+++ CMakeLists.txt	2012-02-01 21:09:20.544138322 -0700
@@ -28,6 +28,14 @@
   unset(WIN32)
 ENDIF()
 
+
+# make sure -fPIC is used, this is a patch provided by openstudio
+if (UNIX)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -fno-strict-aliasing")
+  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC -fno-strict-aliasing")
+endif()
+
+
 # the following lines are for cross compiling support
 # we may get here also from ParaView, in this case don't change the filename
 IF(NOT EXPORT_EXECUTABLES_FILE)
